src/lib.rs

```diff
@@ -315,15 +346,21 @@ pub fn load_and_index_pois(
     };
     let poi_index = rubber.make_index(&dataset, &index_settings).unwrap();
 
-    rows.iterator()
+    let mut pois = rows
+        .iterator()
         .filter_map(|r| {
             r.map_err(|r| warn!("Impossible to load the row {:?}", r))
                 .ok()
         }).filter_map(|p| {
             build_poi(p)
                 .ok_or_else(|| warn!("Problem occurred in build_poi()"))
                 .ok()
-        }).pack(1000)
+        }).collect::<Vec<Poi>>();
```

Comment:
We can't afford to collect all POIs from the database in memory.  
And this is not very useful in this case. The maximum weight is easy to predict, it could directly be normalized in `build_poi`.